packages/typed-message/base/binary-encode/decode.ts

```diff
@@ -49,13 +49,15 @@ function assertString(x: unknown): asserts x is string {
     if (typeof x !== 'string') throw new TypeError(`${HEAD}Invalid TypedMessage`)
 }
 // Detach Uint8Array from it's underlying buffer
-function fixU8Array(obj: unknown) {
+function fixU8Array(input: unknown) {
     // for Array and object
-    if (typeof obj === 'object' && obj !== null) {
-        for (const key in obj) {
-            const val = (obj as any)[key]
-            if (val instanceof Uint8Array) (obj as any)[key] = val.slice()
-            else fixU8Array(val)
+    if (typeof input !== 'object' || input === undefined || input === null) return
+    for (const key of Object.keys(input) as Array<keyof typeof input>) {
+        const value = input[key] as unknown
```

Comment:
You can't get unknown from Object.keys.

Reply:
<img width="424" alt="CleanShot 2022-05-17 at 12 26 17@2x" src="https://user-images.githubusercontent.com/3842474/168728764-35e61252-5fca-4262-9fb1-1eaafdd39cc0.png">